src/identity/contactMapping.ts

```diff
@@ -247,6 +252,54 @@ function* fetchWalletAddresses(e164Number: string) {
   return Array.from(possibleUserAddresses)
 }
 
+function* fetchWalletAddresses(e164Number: string) {
+  const addressesFromDecentralizedMapping: string[] = yield call(
+    fetchWalletAddressesDecentralized,
+    e164Number
+  )
+  const centralPhoneVerificationEnabled = yield select(centralPhoneVerificationEnabledSelector)
+
+  if (!centralPhoneVerificationEnabled) {
+    return addressesFromDecentralizedMapping
+  }
+
+  try {
+    const address = yield select(walletAddressSelector)
+    const signedMessage = yield call(retrieveSignedMessage)
+
+    const response: Response = yield call(fetch, networkConfig.lookupPhoneNumberUrl, {
```

Comment:
Do you think it's worth making the decentralized and centralized fetch in parallel?
Not sure how quick they are today, but could be interesting.